core/domain/stats_jobs_one_off.py

```diff
@@ -345,15 +352,28 @@ def reduce(exp_id, stringified_values):
                 # implicit, pseudo-END state.
                 init_state = versioned_exploration.states[
                     versioned_exploration.init_state_name]
-                max_first_hit_from_init_state = max([
-                    state_stats_mapping[
-                        answer_group.outcome.dest].first_hit_count_v1
-                    for answer_group in init_state.interaction.answer_groups
-                    if answer_group.outcome.dest != (
-                        versioned_exploration.init_state_name) and (
-                            answer_group.outcome.dest in (
-                                versioned_exploration.states))] or [0])
-                num_actual_starts = max_first_hit_from_init_state
+                first_hits_from_init_state = []
+                # log exp_id, exp_version, state_names, state_stats_mapping.
+                for answer_group in init_state.interaction.answer_groups:
+                    dest_state = answer_group.outcome.dest
+                    if dest_state != (
+                            versioned_exploration.init_state_name) and (
+                                dest_state in versioned_exploration.states):
+                        try:
+                            first_hits_from_init_state.append(
+                                state_stats_mapping[
+                                    dest_state].first_hit_count_v1)
+                        except KeyError:
+                            yield (
+                                'ERROR: Pseudo-END state: exp_id %s, version '
+                                '%s, states [%s], state_stats_mapping [%s]' % (
+                                    exp_id, version,
+                                    ', '.join(
+                                        map(str, versioned_exploration.states)),
+                                    ', '.join(
+                                        map(str, state_stats_mapping.keys()))))
```

Comment:
Should we do `raise` after the yield to make sure this still gets treated as an error?

Reply:
Done.